tempesta_fw/http_parser.c

```diff
@@ -866,6 +850,10 @@ __FSM_STATE(RGen_BodyChunkExt) {					\
 	return TFW_BLOCK;						\
 }
 
+#define TFW_HTTP_UPDATE_BODY_IF_OPENED()				\
+	if (msg->body.ptr)						\
+		tfw_http_msg_add_data_ptr(msg, &msg->body, data, len);
```

Comment:
I reviewed the macro more carefully and the comment is right. HTTP parser is a pushdown automaton and we really can check `msg->body.ptr` (as a stack value), but there is no sense to do this for _all_ states. Rather we should do this _not_earlier_ than reach body processing states. So the macro breaks automata concepts.